bin/app.dart

```diff
@@ -0,0 +1,92 @@
+/**
+ * Created by lejard_h on 06/05/16.
+ */
+
+import "dart:io";
+import 'dart:async';
+
+import "package:dev_string_converter/dev_string_converter.dart";
+import "utils.dart";
+
+
+StreamSubscription _progressSubscription;
+
+void _progress() {
+    _progressSubscription =
+        new Stream.periodic(const Duration(seconds: 1)).listen((_) {
+            output('.', Color.white);
+        });
+}
+
+void _endProgress() {
+    _progressSubscription.cancel();
+    stdout.write('\n');
+}
+
+
+main(List<String> args) async {
+    var name = toTableName(args[0]);
+    output('Cloning template repository', Color.green);
+    _progress();
+    ProcessResult result;
+    try {
+        result = await Process.run('git', [
+            'clone',
+            'https://github.com/lejard-h/angular2_project_template.git',
+            '-bmaster',
+            '--single-branch',
+            name,
+        ]);
```

Comment:
only the skeleton is defined, there is no component, only lib. The main things, is that I defined a lot of grind task and the architecture to use dogma-codegen, and I also prepare the template to be used by startrr, 
but yes we can move the template inside ng2gen, it's juste it will be a little bit messy I think.